app/controllers/posts_controller.rb

```diff
@@ -3,6 +3,11 @@ class PostsController < ApplicationController
   skip_authorization_check only: [:create]
   def index
     @posts_by_author = Post.includes(:author).group_by(&:author)
+    serialized_posts = {}
+    @posts_by_author.each do |author, post|
+      serialized_posts[author] = post.as_json(only: %i[Text Title])
+    end
+    render json: serialized_posts
   end
 
   def show
```

Comment:
- [ ] As suggested by the previous reviewer Kindly Fix the the error that is preventing the user from creating a post via API. When I try to create post I got the error as shown with the below screenshhot.
- Creating post 
![image](https://github.com/ismayito/Rails-blog-app/assets/93598189/82287c68-d557-4d02-84c1-cacf71fab43d)

- The error I got 
![image](https://github.com/ismayito/Rails-blog-app/assets/93598189/682b3060-105e-42a1-b5d8-a4f256f09521)

- So to fix the above the issues you are using `CanCanCan` for authorization in your Rails application. If you are encountering an `ActiveModel::ForbiddenAttributesError` in the `create` action, you need to update your **create action to use strong parameters.**